Yoast/wordpress-seo
# Fix vendor prefixing composer

## Summary

This removes all references to the `YoastSEO_Vendor` namespace when creating the composer-artifact.

This also means we no longer need to register our own autoloader to handle situations where classes in the `YoastSEO_Vendor` don't exist as the composer artifact will be directly calling the actual classes.

This PR can be summarized in the following changelog entry:

* Ensures the `YOAST_SEO_EXPERIMENTAL_PHP56` feature flag is compatible with composer installations.

## Relevant technical choices:

To do this https://github.com/Atanamo/PHP-Codeshift is used. However this does require PHP-Parser 4.X and as such PHP_Scoper has also been upgraded, hence the changes in those config files. I've validated the output of both the old and new vendor prefixed directories and they are functionally equal. There's even an extraordinary edge-case handled that would cause issues if another plugin were to use the `YoastSEO_Vendor` as a prefix.

If you wish to test this:
* generate the `vendor_prefixed` directory on trunk and copy it to another directory.
* generate the `vendor_prefixed` directory on this branch.
* Use something like `colordiff -r` to compare the two directories.

## Test instructions
This PR can be tested by following these steps:

* Create a `composer-artifact` on trunk and copy it to another directory.
	* This is done by running `composer install && grunt artifact && grunt composer-artifact`.
* Create a `composer-artifact` on this branch and compare it to the earlier one.
* The only changes should be removals of the `YoastSEO_Vendor` namespace and changes and the changes to the `Dependency_Management` class that are present in this PR.
* There should be no references at all in the code to the `YoastSEO_Vendor` namespace in the `composer-artifact` generated in this branch.

## UI changes

## Documentation

## Quality assurance


Fixes the `YOAST_SEO_EXPERIMENTAL_PHP56` feature flag crashing on composer installations.
